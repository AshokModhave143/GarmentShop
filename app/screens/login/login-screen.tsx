import React, { useState } from 'react'
import { Controller, useForm } from 'react-hook-form'
import { View, Button } from 'react-native'
import { TextInput } from 'react-native-paper'
import { Text } from '../../components'
import { useAppDispatch } from '../../store'
import { login } from '../../store/auth/auth.slice'
import { styles } from './login-screen.style'

export type LoginProps = {
  navigation: any
}
export const LoginScreen: React.FC<LoginProps> = (props: LoginProps) => {
  const [val, setVal] = useState(null)
  const dispatch = useAppDispatch()
  const { control, watch } = useForm()
  const valuse = watch('username')

  return (
    <View style={styles.FULL}>
      <Text style={styles.TEXT}>Login page</Text>
      <Controller
        name="username"
        control={control}
        defaultValue=""
        rules={{
          required: true,
        }}
        render={({ field: { onChange } }) => (
          <TextInput
            label="Username"
            onChangeText={onChange}
            style={styles.TEXT_INPUT}
            placeholder="Enter username"
          />
        )}
      />
      <Button
        title="Click Me"
        onPress={() => {
          console.log(props)
          dispatch(login({ username: 'ashok', password: '1111' }))
        }}
      />
      <Text style={styles.OUTPUT}>{valuse}</Text>
    </View>
  )
}
