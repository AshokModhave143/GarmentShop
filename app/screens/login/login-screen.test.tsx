import React from 'react'
import { render, cleanup } from '../../../test/test-util'
import { LoginScreen } from './login-screen'

afterEach(cleanup)

describe('Login screen', () => {
  it('should render the component without crashing', () => {
    const props = { navigation: { navigate: jest.fn() } }
    const { toJSON } = render(<LoginScreen {...props} />, {})
    expect(toJSON()).toMatchSnapshot()
  })
})
