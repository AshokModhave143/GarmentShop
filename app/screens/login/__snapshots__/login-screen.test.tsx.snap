// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen should render the component without crashing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
          },
          Object {
            "color": "#1d1d1d",
            "fontFamily": "Roboto-Regular",
            "marginHorizontal": 8,
          },
        ]
      }
    >
      Hello
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(231, 231, 231)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          Object {},
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.26)",
            "bottom": 0,
            "height": 2,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": Array [
              Object {
                "scaleY": 0.5,
              },
            ],
            "zIndex": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "minHeight": 80,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingLeft": 12,
                "paddingRight": 12,
                "position": "absolute",
                "textAlign": "left",
                "top": 40,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Username
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingLeft": 12,
                "paddingRight": 12,
                "position": "absolute",
                "textAlign": "left",
                "top": 40,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Username
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
              },
              Object {
                "paddingLeft": 12,
                "paddingRight": 12,
              },
              Object {
                "height": 80,
              },
              Object {
                "paddingBottom": 4,
                "paddingTop": 24,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Click Me
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
          },
          Object {
            "color": "#FBA928",
          },
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
